### DISCUSSION TITLE
Service Account Access

### DISCUSSION THREAD

**Discussion by cookrn** [nytimes/library#264] [Q&A/Help]
Hello! I've been working on getting an instance of Library running locally prior to deploying in a test environment and think I have it mostly working except for one part. At the bottom of [this page](https://nyt-library-demo.herokuapp.com/get-started/create-a-service-account) when attempting to share the folder with the service account address, I don't have that option. Instead, "Share" is disabled and grayed out.

After reading a bit, I found this [Google support answer](https://support.google.com/drive/answer/7166529?co=GENIE.Platform%3DDesktop&hl=en) which says `Note: You may not share folders stored in Team Drives.`

We attempted to give the service account read only access to the team drive even though we're using a sub-folder, but no luck.

Here's the log output when attempting to load `localhost:3000`:

```
app_1  | debug: searching for files > 0
app_1  | debug: Current file count in drive: 0
app_1  | debug: tree updated.
app_1  | error: Serving an error page for /
app_1  | { message: 'Cannot convert undefined or null to object',
app_1  |   stack:
app_1  |    'TypeError: Cannot convert undefined or null to object\n    at Function.keys (<anonymous>)\n    at buildDisplayCategories (/usr/src/app/server/routes/pages.js:56:29)\n    at handlePage (/usr/src/app/server/routes/pages.js:47:24)' }
```

It may be unrelated since we aren't customizing any of the behavior, but I see the following debug errors during boot:

> debug: Failed pulling in custom file cache/store @ /usr/src/app/custom/cache/store
> debug: Failed pulling in custom file userAuth @ /usr/src/app/custom/userAuth
> debug: Failed pulling in custom file csp @ /usr/src/app/custom/csp

Maybe we need to specify these? Is there a different way to give the service account access to a subfolder of our team drive that we might be missing? Thanks in advance!

**Comment by isaacwhite** [comment]
Hey @cookrn, 

As you mentioned, it's not possible to share subfolder within team drives. Instead, you need to share access on the entire team drive or use a folder outside the team drive to share that content as a blanket rule.

The first several debug log lines above indicate that the service account didn't find any Google Docs within your shared folder to display in Library. That's likely the cause of the error on the following lines `Cannot convert undefined or null to object`. 

So far as the debug lines about failing to pull in customization options for those files go, they should not affect your ability to run the app. Are there documents within the subfolder you tried to share with Library? If so, it may indicate that Library doesn't work yet pointed at a subfolder of a Team Drive. 

We've never tested that configuration before, but it sounds like a valid use case to me and we could open a new issue to support that if it's not possible currently.
